include/envoy/init/target.h

```diff
@@ -25,6 +25,11 @@ struct TargetHandle {
    * @return true if the target received this call, false if the target was already destroyed.
    */
   virtual bool initialize(const Watcher& watcher) const PURE;
+
+  /**
+   * @return a human-readable target name, for logging / debugging.
```

Comment:
This is not only for logging / debugging but for target tracking as well? Comment it it have to be unique (otherwise the ASSERT below can fail).